### PR TITLE
Make insertedWith const and update unit tests

### PR DESCRIPTION
**Pull request by jonasseglare**
Just some fixes.

### CODE REVIEW

**Comment by cnd** [comment]
please don't mix tabs and spaces

File: RBMap/MapTest.cpp [hunk 1]
```diff
@@ -17,4 +23,14 @@ void main()
     std::vector<std::pair<char, std::string>> v = { { 'c', "cockney" }, { 'a', "abba" }, { 'b', "bbc" } };
     auto map2 = fromListOfPairs<char, std::string>(v.begin(), v.end());
     std::cout << map2 << std::endl;
+
+		auto map3 = map.inserted(3, "three").inserted(3, "iii");
```
Comment (jonasseglare) [inline RBMap/MapTest.cpp]: Fix indentation issues here.

**Comment by cnd** [comment]
thank you, and line 10...

**Comment by jonasseglare** [comment]
I believe formatting is pretty good now.

**Pull request merged**